### PR TITLE
Add focus states to boolean and checkbox

### PR DESCRIPTION
**Pull request by andrico1234**
Fixes: https://github.com/marmelab/react-admin/issues/7771

https://user-images.githubusercontent.com/17087167/171428166-a9e90c03-c313-49a2-878f-3ac9114ab5dc.mov
https://user-images.githubusercontent.com/17087167/171428222-c88c5c02-646b-415b-b789-ccbaecaa57a2.mov

### CODE REVIEW

**Comment by fzaninotto** [comment]
absolutely!

File: docs/Buttons.md [hunk 1]
```diff
@@ -307,6 +307,16 @@ const MyEdit = () => (
 
 ### `<CloneButton>`
 
+## Performance
```
Comment (andrico1234) [inline docs/Buttons.md:310]: Wasn't sure whether to have this in the button or datagrid component docs, or to create a new top-level page specifically for performance. Happy to go with whatever you folks feel is most appropriate

**Comment by andrico1234** [comment]
friendly bump @fzaninotto

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**